tests/typing/faulty_if_expressions.sy

```diff
@@ -1,8 +0,0 @@
-start :: fn do
-    1.0 + 1.0 if true else 2.0
-    (1.0 if < 0.0 else 2.0) + 1.0
-    1.0 + 1.0 if "potato" else 2.0
-    (1.0 if < "abc" else 2.0) + 1.0
-end
-// error: $Mismatch { expected: Type::Bool, .. }
-// error: $BinOp { lhs: Type::Float, .. }
```

Comment:
These are still valid tests right? We should keep them (with some modifications).

Reply:
No - these are not valid. They tested a syntax and semantic that has been removed - the short-if-statements.